// angular-gallery/esm2015/lib/overlay/overlay.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { EventService } from './event.service';
import { Overlay } from './overlay.service';
import { OverlayContainerComponent } from './overlay-container.component';
import { AdDirective } from './ad.directive';
import * as ɵngcc0 from '@angular/core';
let OverlayModule = class OverlayModule {
};
OverlayModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: OverlayModule });
OverlayModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function OverlayModule_Factory(t) { return new (t || OverlayModule)(); }, providers: [
        Overlay,
        EventService
    ], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(OverlayModule, { declarations: function () { return [OverlayContainerComponent,
        AdDirective]; }, imports: function () { return [CommonModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OverlayModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    OverlayContainerComponent,
                    AdDirective
                ],
                imports: [
                    CommonModule
                ],
                exports: [],
                providers: [
                    Overlay,
                    EventService
                ],
                bootstrap: [],
                entryComponents: [
                    OverlayContainerComponent
                ]
            }]
    }], null, null); })();
export { OverlayModule };
