// angular-gallery/esm2015/lib/carousel.component.js
import { __decorate } from "tslib";
import { ChangeDetectorRef, Component, ElementRef, EventEmitter, HostBinding, HostListener, Input, Output, OnDestroy, SimpleChanges } from '@angular/core';
import { Touches } from './touches';
import { Carousel } from './carousel';
import { EventService as OverlayEventService } from './overlay/event.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './overlay/event.service';
import * as ɵngcc2 from '@angular/common';

function CarouselComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 7);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.counter);
} }
function CarouselComponent_ng_template_5_div_0_img_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 12);
} if (rf & 2) {
    const i_r4 = ɵngcc0.ɵɵnextContext(2).index;
    const ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("object-fit", ctx_r6.objectFit);
    ɵngcc0.ɵɵproperty("src", ctx_r6.getFile(i_r4)["image"]["path"], ɵngcc0.ɵɵsanitizeUrl);
} }
function CarouselComponent_ng_template_5_div_0_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    const _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 13);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_ng_template_5_div_0_ng_template_2_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r11); const _r9 = ɵngcc0.ɵɵreference(2); const ctx_r10 = ɵngcc0.ɵɵnextContext(3); return ctx_r10.toggleVideo(_r9); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(1, "video", 14, 15);
    ɵngcc0.ɵɵelement(3, "source", 16);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const i_r4 = ɵngcc0.ɵɵnextContext(2).index;
    const ctx_r7 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleProp("object-fit", ctx_r7.objectFit)("border-radius", ctx_r7.borderRadius + "px");
    ɵngcc0.ɵɵproperty("loop", ctx_r7.videoProperties.loop)("muted", ctx_r7.videoProperties.muted)("controls", ctx_r7.isSafari);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("src", ctx_r7.getFile(i_r4)["path"], ɵngcc0.ɵɵsanitizeUrl);
} }
function CarouselComponent_ng_template_5_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵtemplate(1, CarouselComponent_ng_template_5_div_0_img_1_Template, 1, 3, "img", 10);
    ɵngcc0.ɵɵtemplate(2, CarouselComponent_ng_template_5_div_0_ng_template_2_Template, 4, 8, "ng-template", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const i_r4 = ɵngcc0.ɵɵnextContext().index;
    const ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("width", ctx_r5.getCellWidth() + "px")("border-radius", ctx_r5.borderRadius + "px");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r5.getFile(i_r4) && ctx_r5.getFile(i_r4)["image"]);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r5.getFile(i_r4) && ctx_r5.getFile(i_r4)["type"] === "video");
} }
function CarouselComponent_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, CarouselComponent_ng_template_5_div_0_Template, 3, 6, "div", 8);
} if (rf & 2) {
    const i_r4 = ctx.index;
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", i_r4 < ctx_r1.cellLimit);
} }
function CarouselComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    const _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 17);
    ɵngcc0.ɵɵelementStart(1, "div", 18);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_div_6_Template_div_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r15); const ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.prev(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(2, "div", 19);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_div_6_Template_div_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r15); const ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.next(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("carousel-arrows-outside", ctx_r2.arrowsOutside)("carousel-dark-arrows", ctx_r2.arrowsTheme === "dark");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassProp("carousel-arrow-disabled", ctx_r2.isPrevArrowDisabled());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassProp("carousel-arrow-disabled", ctx_r2.isNextArrowDisabled());
} }
let CarouselComponent = class CarouselComponent {
    constructor(elementRef, ref, overlayEventService) {
        this.elementRef = elementRef;
        this.ref = ref;
        this.overlayEventService = overlayEventService;
        this.minTimeout = 30;
        this._cellWidth = 200;
        this._transitionDuration = 200;
        this.events = new EventEmitter();
        this.height = 200;
        this.margin = 10;
        this.objectFit = 'cover';
        this.minSwipeDistance = 50;
        this.transitionTimingFunction = 'ease';
        this.counterSeparator = " / ";
        this.overflowCellsLimit = 3;
        this.listeners = 'mouse and touch';
        this.arrows = true;
        this.arrowsTheme = 'light';
        this.hostClassCarousel = true;
        /* Touchstart */
        this.handleTouchstart = (event) => {
            event.preventDefault();
            this.carousel.handleTouchstart(event);
            this.isMoving = true;
            this.events.emit({
                type: 'touchstart',
                event
            });
        };
        /* Touchmove */
        this.handleHorizontalSwipe = (event) => {
            event.preventDefault();
            this.carousel.handleHorizontalSwipe(event);
            this.events.emit({
                type: 'swipe',
                event
            });
        };
        /* Touchend */
        this.handleTouchend = (event) => {
            const touches = event.touches;
            this.carousel.handleTouchend(event);
            this.isMoving = false;
            this.events.emit({
                type: 'touchend',
                event
            });
        };
        /* Tap */
        this.handleTap = (event) => {
            const i = this.carousel.slideCounter;
            const cellIndex = this.carousel.currentCellIndex;
            const fileIndex = this.carousel.getFileIndex(i);
            const file = this.carousel.getFile(cellIndex);
            this.events.emit({
                type: 'click',
                file: file,
                index: fileIndex
            });
        };
    }
    get isLandscape() {
        return window.innerWidth > window.innerHeight;
    }
    get isSafari() {
        const ua = navigator.userAgent.toLowerCase();
        if (ua.indexOf('safari') !== -1) {
            return !(ua.indexOf('chrome') > -1);
        }
    }
    get counter() {
        return this.carousel.slideCounter + 1 + this.counterSeparator + this.images.length;
    }
    set images(images) {
        this._images = images;
    }
    get images() {
        return this._images;
    }
    get isNgContent() {
        return this.elementRef.nativeElement.querySelector('.carousel-content-wrapper').children.length > 0;
    }
    set transitionDuration(value) {
        this._transitionDuration = value;
    }
    get transitionDuration() {
        if (this._transitionDuration === 0) {
            return 1;
        }
        else {
            return this._transitionDuration;
        }
    }
    set cellWidth(value) {
        if (value) {
            this._cellWidth = value;
        }
    }
    set isCounter(value) {
        if (value) {
            this._isCounter = value;
        }
    }
    get isCounter() {
        return this._isCounter && this.images.length > 1;
    }
    set overlayProperties(properties) {
        this._properties = properties;
        this._images = this._properties.metadata.images;
        this._cellWidth = this._properties.metadata.width;
        this.height = this._properties.metadata.height;
        this.objectFit = this._properties.metadata.objectFit;
        this.margin = this._properties.metadata.margin;
        this.transitionDuration = this._properties.metadata.transitionDuration;
        this.overflowCellsLimit = this._properties.metadata.overflowCellsLimit;
        this.minSwipeDistance = this._properties.metadata.minSwipeDistance;
        this.transitionTimingFunction = this._properties.metadata.transitionTimingFunction;
        this._isCounter = this._properties.metadata.counter;
        this.counterSeparator = this._properties.metadata.counterSeparator;
        this.arrows = this._properties.metadata.arrows;
    }
    get properties() {
        return this._properties;
    }
    get cellLimit() {
        if (this.carousel) {
            return this.carousel.cellLimit;
        }
    }
    onWindowResize(event) {
        this.landscapeMode = this.isLandscape;
        this.ref.detectChanges();
        this.initCarousel();
        this.carousel.lineUpCells();
    }
    ngOnInit() {
        this.touches = new Touches({
            element: this.elementRef.nativeElement.querySelector('.carousel-cells'),
            listeners: this.listeners
        });
        this.touches.on('touchstart', this.handleTouchstart);
        this.touches.on('horizontal-swipe', this.handleHorizontalSwipe);
        this.touches.on('touchend', this.handleTouchend);
        this.touches.on('mousedown', this.handleTouchstart);
        this.touches.on('mouseup', this.handleTouchend);
        this.touches.on('tap', this.handleTap);
        this.initCarousel();
        this.setDimensions();
        this.setInitialIndex();
    }
    ngAfterViewInit() {
        this.carousel.lineUpCells();
    }
    ngOnChanges(changes) {
        if (changes.width || changes.height) {
            this.setDimensions();
            this.initCarousel();
            this.carousel.lineUpCells();
        }
    }
    ngOnDestroy() {
        this.touches.destroy();
    }
    initCarousel() {
        this.carousel = new Carousel({
            element: this.elementRef.nativeElement.querySelector('.carousel-cells'),
            container: this.elementRef.nativeElement,
            images: this.images,
            cellWidth: this.getCellWidth(),
            overflowCellsLimit: this.overflowCellsLimit,
            visibleWidth: this.width,
            margin: this.margin,
            minSwipeDistance: this.minSwipeDistance,
            transitionDuration: this.transitionDuration,
            transitionTimingFunction: this.transitionTimingFunction,
            videoProperties: this.videoProperties
        });
    }
    setDimensions() {
        this.hostStyleHeight = this.height + 'px';
        this.hostStyleWidth = this.width + 'px';
    }
    getFile(index) {
        return this.carousel.getFile(index);
    }
    handleTransitionendCellContainer(event) {
        this.carousel.handleSlideEnd();
    }
    toggleVideo(video) {
        event.preventDefault();
        if (this.videoProperties.noPlay) {
            return;
        }
        if (video.paused) {
            video.play();
            this.isVideoPlaying = true;
        }
        else {
            video.pause();
            this.isVideoPlaying = false;
        }
        this.ref.detectChanges();
    }
    getCurrentIndex() {
        return this.carousel.slideCounter;
    }
    getCellWidth() {
        if (this._cellWidth === '100%') {
            return this.elementRef.nativeElement.clientWidth;
        }
        else {
            return this._cellWidth;
        }
    }
    next() {
        this.carousel.next(1);
    }
    prev() {
        this.carousel.prev(1);
    }
    select(index) {
        this.carousel.select(index);
    }
    isNextArrowDisabled() {
        return this.carousel.isNextArrowDisabled();
    }
    isPrevArrowDisabled() {
        return this.carousel.isPrevArrowDisabled();
    }
    close() {
        this.overlayEventService.emitChangeEvent({
            type: 'Hide'
        });
    }
    setInitialIndex() {
        if (this._properties.metadata.index != undefined) {
            this.select(this._properties.metadata.index);
        }
    }
};
CarouselComponent.ɵfac = function CarouselComponent_Factory(t) { return new (t || CarouselComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.EventService)); };
CarouselComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselComponent, selectors: [["carousel"], ["", "carousel", ""]], hostVars: 6, hostBindings: function CarouselComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("resize", function CarouselComponent_resize_HostBindingHandler($event) { return ctx.onWindowResize($event); }, false, ɵngcc0.ɵɵresolveWindow);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("height", ctx.hostStyleHeight)("width", ctx.hostStyleWidth);
        ɵngcc0.ɵɵclassProp("carousel", ctx.hostClassCarousel);
    } }, inputs: { height: "height", margin: "margin", objectFit: "objectFit", minSwipeDistance: "minSwipeDistance", transitionTimingFunction: "transitionTimingFunction", counterSeparator: "counterSeparator", overflowCellsLimit: "overflowCellsLimit", listeners: "listeners", arrows: "arrows", arrowsTheme: "arrowsTheme", images: "images", transitionDuration: "transitionDuration", cellWidth: "cellWidth", isCounter: ["counter", "isCounter"], overlayProperties: "overlayProperties", width: "width", borderRadius: "borderRadius", videoProperties: "videoProperties", arrowsOutside: "arrowsOutside" }, outputs: { events: "events" }, exportAs: ["carousel"], features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 7, vars: 5, consts: [[1, "carousel-backdrop"], [1, "carousel-close", 3, "click"], ["class", "carousel-counter", 4, "ngIf"], [1, "carousel-container"], [1, "carousel-cells", 3, "transitionend"], ["ngFor", "", 3, "ngForOf"], ["class", "carousel-arrows", 3, "carousel-arrows-outside", "carousel-dark-arrows", 4, "ngIf"], [1, "carousel-counter"], ["class", "carousel-cell", 3, "width", "border-radius", 4, "ngIf"], [1, "carousel-cell"], ["draggable", "false", 3, "src", "object-fit", 4, "ngIf"], [3, "ngIf"], ["draggable", "false", 3, "src"], [1, "carousel-play", 3, "click"], [3, "loop", "muted", "controls"], ["video", ""], ["type", "video/mp4", 3, "src"], [1, "carousel-arrows"], [1, "carousel-arrow", "carousel-arrow-prev", 3, "click"], [1, "carousel-arrow", "carousel-arrow-next", 3, "click"]], template: function CarouselComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵlistener("click", function CarouselComponent_Template_div_click_1_listener() { return ctx.close(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, CarouselComponent_div_2_Template, 2, 1, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵelementStart(4, "div", 4);
        ɵngcc0.ɵɵlistener("transitionend", function CarouselComponent_Template_div_transitionend_4_listener($event) { return ctx.handleTransitionendCellContainer($event); });
        ɵngcc0.ɵɵtemplate(5, CarouselComponent_ng_template_5_Template, 1, 1, "ng-template", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, CarouselComponent_div_6_Template, 3, 8, "div", 6);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isCounter);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassProp("carousel-moving", ctx.isMoving);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.images);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.arrows);
    } }, directives: [ɵngcc2.NgIf, ɵngcc2.NgForOf], styles: ["[_nghost-%COMP%]{position:relative;display:block;top:0;left:0;width:100%;height:100%;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;z-index:10000;transform-origin:top left;box-sizing:border-box}[_nghost-%COMP%]   .carousel-container[_ngcontent-%COMP%]{overflow:hidden;width:100%;height:100%;cursor:-webkit-grab;cursor:grab}[_nghost-%COMP%]   .carousel-container.carousel-moving[_ngcontent-%COMP%]{cursor:-webkit-grabbing;cursor:grabbing}[_nghost-%COMP%]   .carousel-counter[_ngcontent-%COMP%]{text-align:right;position:absolute;z-index:30;transition:opacity .2s;top:15px;left:10px;border-radius:13px;background-color:rgba(23,37,68,.3);font-size:13px;color:#fff;padding:5px 7px;line-height:initial}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]{transition:transform .2s;width:100%;height:100%;display:block}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell.swiper-prev-image{transform:translate3d(-100%,0,0)}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell.swiper-next-image{transform:translate3d(100%,0,0)}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell{width:100%;height:100%;text-align:center;position:absolute;display:flex;align-items:center;justify-content:center;overflow:hidden}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell img, [_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell video{width:100%;height:100%;position:relative;-o-object-fit:contain;object-fit:contain}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell img.swiper-hide{display:none}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell .carousel-play{position:absolute;top:0;left:0;bottom:0;right:0;z-index:1}[_nghost-%COMP%]   .carousel-arrow[_ngcontent-%COMP%]{width:40px;height:40px;background-color:#fff;background-repeat:no-repeat;background-size:31px;background-position:center;border-radius:100px;position:absolute;top:50%;margin-top:-20px;z-index:10;cursor:pointer;box-shadow:0 0 5px rgba(0,0,0,.15)}[_nghost-%COMP%]   .carousel-arrow-prev[_ngcontent-%COMP%]{left:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTUuNDEgMTYuNTlMMTAuODMgMTJsNC41OC00LjU5TDE0IDZsLTYgNiA2IDYgMS40MS0xLjQxeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMFYweiIgZmlsbD0ibm9uZSIvPjwvc3ZnPg==)}[_nghost-%COMP%]   .carousel-arrow-next[_ngcontent-%COMP%]{right:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNOC41OSAxNi41OUwxMy4xNyAxMiA4LjU5IDcuNDEgMTAgNmw2IDYtNiA2LTEuNDEtMS40MXoiLz48cGF0aCBkPSJNMCAwaDI0djI0SDBWMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=)}[_nghost-%COMP%]   .carousel-arrows-outside[_ngcontent-%COMP%]   .carousel-arrow-prev[_ngcontent-%COMP%]{left:-60px}[_nghost-%COMP%]   .carousel-arrows-outside[_ngcontent-%COMP%]   .carousel-arrow-next[_ngcontent-%COMP%]{right:-60px}[_nghost-%COMP%]   .carousel-dark-arrows[_ngcontent-%COMP%]   .carousel-arrow[_ngcontent-%COMP%]{-webkit-filter:invert(1);filter:invert(1)}[_nghost-%COMP%]   .carousel-arrow-disabled[_ngcontent-%COMP%]{cursor:default;opacity:.5}[_nghost-%COMP%]   .carousel-close[_ngcontent-%COMP%]{position:fixed;top:10px;right:10px;width:40px;height:40px;z-index:100;cursor:pointer;background-repeat:no-repeat;background-size:31px;background-position:center;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTkgNi40MUwxNy41OSA1IDEyIDEwLjU5IDYuNDEgNSA1IDYuNDEgMTAuNTkgMTIgNSAxNy41OSA2LjQxIDE5IDEyIDEzLjQxIDE3LjU5IDE5IDE5IDE3LjU5IDEzLjQxIDEyeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=);-webkit-filter:invert(1);filter:invert(1)}[_nghost-%COMP%]   .carousel-backdrop[_ngcontent-%COMP%]{position:fixed;top:0;left:0;right:0;bottom:0;background-color:#000;opacity:.5}"] });
CarouselComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef },
    { type: OverlayEventService }
];
__decorate([
    Input()
], CarouselComponent.prototype, "images", null);
__decorate([
    Output()
], CarouselComponent.prototype, "events", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "height", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "width", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "borderRadius", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "margin", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "objectFit", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "minSwipeDistance", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "transitionDuration", null);
__decorate([
    Input()
], CarouselComponent.prototype, "transitionTimingFunction", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "videoProperties", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "counterSeparator", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "overflowCellsLimit", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "listeners", void 0);
__decorate([
    Input('cellWidth')
], CarouselComponent.prototype, "cellWidth", null);
__decorate([
    Input('counter')
], CarouselComponent.prototype, "isCounter", null);
__decorate([
    Input()
], CarouselComponent.prototype, "arrows", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "arrowsOutside", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "arrowsTheme", void 0);
__decorate([
    Input()
], CarouselComponent.prototype, "overlayProperties", null);
__decorate([
    HostBinding('class.carousel')
], CarouselComponent.prototype, "hostClassCarousel", void 0);
__decorate([
    HostBinding('style.height')
], CarouselComponent.prototype, "hostStyleHeight", void 0);
__decorate([
    HostBinding('style.width')
], CarouselComponent.prototype, "hostStyleWidth", void 0);
__decorate([
    HostListener('window:resize', ['$event'])
], CarouselComponent.prototype, "onWindowResize", null);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CarouselComponent, [{
        type: Component,
        args: [{
                selector: 'carousel, [carousel]',
                exportAs: 'carousel',
                template: "<div class=\"carousel-backdrop\"></div>\n\n<div class=\"carousel-close\" (click)=\"close()\"></div>\n\n<div class=\"carousel-counter\" *ngIf=\"isCounter\">{{counter}}</div>\n\n<div class=\"carousel-container\" [class.carousel-moving]=\"isMoving\">\n\t<div class=\"carousel-cells\" (transitionend)=\"handleTransitionendCellContainer($event)\">\n\t\t<ng-template ngFor let-image [ngForOf]=\"images\" let-i=\"index\">\n\t\t\t<div class=\"carousel-cell\" \n\t\t\t\t*ngIf=\"i < cellLimit\" \n\t\t\t\t[style.width]=\"getCellWidth()+'px'\"\n\t\t\t\t[style.border-radius]=\"borderRadius+'px'\">\n\t\t\t\t<!-- Image -->\n\t\t\t\t<img \n\t\t\t\t\t*ngIf=\"getFile(i) && getFile(i)['image']\" \n\t\t\t\t\t[src]=\"getFile(i)['image']['path']\"\n\t\t\t\t\t[style.object-fit]=\"objectFit\"\n\t\t\t\t\tdraggable=\"false\" />\n\n\t\t\t\t<!-- Video -->\n\t\t\t\t<ng-template [ngIf]=\"getFile(i) && getFile(i)['type'] === 'video'\">\n\t\t\t\t\t<div class=\"carousel-play\" (click)=\"toggleVideo(video)\"></div>\n\n\t\t\t\t\t<video #video\n\t\t\t\t\t\t[loop]=\"videoProperties.loop\"\n\t\t\t\t\t\t[muted]=\"videoProperties.muted\"\n\t\t\t\t\t\t[controls]=\"isSafari\"\n\t\t\t\t\t\t[style.object-fit]=\"objectFit\"\n\t\t\t\t\t\t[style.border-radius]=\"borderRadius+'px'\">\n\t\t\t\t\t\t\n\t\t\t\t\t\t<source [src]=\"getFile(i)['path']\" type=\"video/mp4\">\n\t\t\t\t\t</video>\n\t\t\t\t</ng-template>\n\t\t\t</div>\n\t\t</ng-template>\n\t</div>\n</div>\n\n<div class=\"carousel-arrows\" \n\t[class.carousel-arrows-outside]=\"arrowsOutside\" \n\t[class.carousel-dark-arrows]=\"arrowsTheme === 'dark'\"\n\t*ngIf=\"arrows\">\n\t\n\t<div class=\"carousel-arrow carousel-arrow-prev\" [class.carousel-arrow-disabled]=\"isPrevArrowDisabled()\" (click)=\"prev()\"></div>\n\t<div class=\"carousel-arrow carousel-arrow-next\" [class.carousel-arrow-disabled]=\"isNextArrowDisabled()\" (click)=\"next()\"></div>\n</div>",
                styles: [":host{position:relative;display:block;top:0;left:0;width:100%;height:100%;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;z-index:10000;transform-origin:top left;box-sizing:border-box}:host .carousel-container{overflow:hidden;width:100%;height:100%;cursor:-webkit-grab;cursor:grab}:host .carousel-container.carousel-moving{cursor:-webkit-grabbing;cursor:grabbing}:host .carousel-counter{text-align:right;position:absolute;z-index:30;transition:opacity .2s;top:15px;left:10px;border-radius:13px;background-color:rgba(23,37,68,.3);font-size:13px;color:#fff;padding:5px 7px;line-height:initial}:host .carousel-cells{transition:transform .2s;width:100%;height:100%;display:block}:host .carousel-cells ::ng-deep .carousel-cell.swiper-prev-image{transform:translate3d(-100%,0,0)}:host .carousel-cells ::ng-deep .carousel-cell.swiper-next-image{transform:translate3d(100%,0,0)}:host .carousel-cells ::ng-deep .carousel-cell{width:100%;height:100%;text-align:center;position:absolute;display:flex;align-items:center;justify-content:center;overflow:hidden}:host .carousel-cells ::ng-deep .carousel-cell img,:host .carousel-cells ::ng-deep .carousel-cell video{width:100%;height:100%;position:relative;-o-object-fit:contain;object-fit:contain}:host .carousel-cells ::ng-deep .carousel-cell img.swiper-hide{display:none}:host .carousel-cells ::ng-deep .carousel-cell .carousel-play{position:absolute;top:0;left:0;bottom:0;right:0;z-index:1}:host .carousel-arrow{width:40px;height:40px;background-color:#fff;background-repeat:no-repeat;background-size:31px;background-position:center;border-radius:100px;position:absolute;top:50%;margin-top:-20px;z-index:10;cursor:pointer;box-shadow:0 0 5px rgba(0,0,0,.15)}:host .carousel-arrow-prev{left:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTUuNDEgMTYuNTlMMTAuODMgMTJsNC41OC00LjU5TDE0IDZsLTYgNiA2IDYgMS40MS0xLjQxeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMFYweiIgZmlsbD0ibm9uZSIvPjwvc3ZnPg==)}:host .carousel-arrow-next{right:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNOC41OSAxNi41OUwxMy4xNyAxMiA4LjU5IDcuNDEgMTAgNmw2IDYtNiA2LTEuNDEtMS40MXoiLz48cGF0aCBkPSJNMCAwaDI0djI0SDBWMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=)}:host .carousel-arrows-outside .carousel-arrow-prev{left:-60px}:host .carousel-arrows-outside .carousel-arrow-next{right:-60px}:host .carousel-dark-arrows .carousel-arrow{-webkit-filter:invert(1);filter:invert(1)}:host .carousel-arrow-disabled{cursor:default;opacity:.5}:host .carousel-close{position:fixed;top:10px;right:10px;width:40px;height:40px;z-index:100;cursor:pointer;background-repeat:no-repeat;background-size:31px;background-position:center;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTkgNi40MUwxNy41OSA1IDEyIDEwLjU5IDYuNDEgNSA1IDYuNDEgMTAuNTkgMTIgNSAxNy41OSA2LjQxIDE5IDEyIDEzLjQxIDE3LjU5IDE5IDE5IDE3LjU5IDEzLjQxIDEyeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=);-webkit-filter:invert(1);filter:invert(1)}:host .carousel-backdrop{position:fixed;top:0;left:0;right:0;bottom:0;background-color:#000;opacity:.5}"]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.ChangeDetectorRef }, { type: ɵngcc1.EventService }]; }, { events: [{
            type: Output
        }], height: [{
            type: Input
        }], margin: [{
            type: Input
        }], objectFit: [{
            type: Input
        }], minSwipeDistance: [{
            type: Input
        }], transitionTimingFunction: [{
            type: Input
        }], counterSeparator: [{
            type: Input
        }], overflowCellsLimit: [{
            type: Input
        }], listeners: [{
            type: Input
        }], arrows: [{
            type: Input
        }], arrowsTheme: [{
            type: Input
        }], hostClassCarousel: [{
            type: HostBinding,
            args: ['class.carousel']
        }], images: [{
            type: Input
        }], transitionDuration: [{
            type: Input
        }], cellWidth: [{
            type: Input,
            args: ['cellWidth']
        }], isCounter: [{
            type: Input,
            args: ['counter']
        }], overlayProperties: [{
            type: Input
        }], onWindowResize: [{
            type: HostListener,
            args: ['window:resize', ['$event']]
        }], hostStyleHeight: [{
            type: HostBinding,
            args: ['style.height']
        }], hostStyleWidth: [{
            type: HostBinding,
            args: ['style.width']
        }], width: [{
            type: Input
        }], borderRadius: [{
            type: Input
        }], videoProperties: [{
            type: Input
        }], arrowsOutside: [{
            type: Input
        }] }); })();
export { CarouselComponent };
