// angular-gallery/esm2015/lib/gallery.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { CarouselComponent } from './carousel.component';
import { Overlay } from './overlay/overlay.service';
import { EventService as OverlayEventService } from './overlay/event.service';
import { defaultProperties } from './default-properties';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './overlay/overlay.service';
import * as ɵngcc2 from './overlay/event.service';
let Gallery = class Gallery {
    constructor(overlay, overlayEventService) {
        this.overlay = overlay;
        this.overlayEventService = overlayEventService;
    }
    load(properties) {
        properties = this.applyPropertieDefaults(defaultProperties, properties);
        this.overlay.load({
            id: 'gallery',
            mainComponent: CarouselComponent,
            width: document.body.clientWidth + 'px',
            overlayBackdrop: true,
            position: 'fixed',
            metadata: {
                images: properties.images,
                index: properties.index,
                width: document.body.clientWidth,
                height: window.innerHeight,
                objectFit: 'none',
                margin: 0,
                transitionDuration: properties.transitionDuration,
                overflowCellsLimit: 1,
                transitionTimingFunction: properties.transitionTimingFunction,
                counter: properties.counter,
                counterSeparator: properties.counterSeparator,
                arrows: properties.arrows,
                minSwipeDistance: properties.minSwipeDistance
            }
        });
    }
    close() {
        this.overlayEventService.emitChangeEvent({
            type: 'Hide'
        });
    }
    applyPropertieDefaults(defaultProperties, properties) {
        if (!properties) {
            properties = {};
        }
        for (var propertie in properties) {
            if (properties[propertie] === undefined) {
                delete properties[propertie];
            }
        }
        this._defaultProperties = Object.assign({}, defaultProperties);
        return Object.assign(this._defaultProperties, properties);
    }
};
Gallery.ɵfac = function Gallery_Factory(t) { return new (t || Gallery)(ɵngcc0.ɵɵinject(ɵngcc1.Overlay), ɵngcc0.ɵɵinject(ɵngcc2.EventService)); };
Gallery.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: Gallery, factory: function (t) { return Gallery.ɵfac(t); } });
Gallery.ctorParameters = () => [
    { type: Overlay },
    { type: OverlayEventService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Gallery, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.Overlay }, { type: ɵngcc2.EventService }]; }, null); })();
export { Gallery };
