// angular-gallery/bundles/angular-gallery.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common'), require('rxjs')) :
    typeof define === 'function' && define.amd ? define('angular-gallery', ['@angular/core','@angular/common','exports', '@angular/core', '@angular/common', 'rxjs'], factory) :
    (global = global || self, factory(global.ng.core,global.ng.common,global['angular-gallery'] = {}, global.ng.core, global.ng.common, global.rxjs));
}(this, (function (ɵngcc0,ɵngcc1,exports, core, common, rxjs) { 
function CarouselComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 7);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.counter);
} }
function CarouselComponent_ng_template_5_div_0_img_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 12);
} if (rf & 2) {
    var i_r4 = ɵngcc0.ɵɵnextContext(2).index;
    var ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("object-fit", ctx_r6.objectFit);
    ɵngcc0.ɵɵproperty("src", ctx_r6.getFile(i_r4)["image"]["path"], ɵngcc0.ɵɵsanitizeUrl);
} }
function CarouselComponent_ng_template_5_div_0_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    var _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 13);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_ng_template_5_div_0_ng_template_2_Template_div_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r11); var _r9 = ɵngcc0.ɵɵreference(2); var ctx_r10 = ɵngcc0.ɵɵnextContext(3); return ctx_r10.toggleVideo(_r9); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(1, "video", 14, 15);
    ɵngcc0.ɵɵelement(3, "source", 16);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var i_r4 = ɵngcc0.ɵɵnextContext(2).index;
    var ctx_r7 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleProp("object-fit", ctx_r7.objectFit)("border-radius", ctx_r7.borderRadius + "px");
    ɵngcc0.ɵɵproperty("loop", ctx_r7.videoProperties.loop)("muted", ctx_r7.videoProperties.muted)("controls", ctx_r7.isSafari);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("src", ctx_r7.getFile(i_r4)["path"], ɵngcc0.ɵɵsanitizeUrl);
} }
function CarouselComponent_ng_template_5_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵtemplate(1, CarouselComponent_ng_template_5_div_0_img_1_Template, 1, 3, "img", 10);
    ɵngcc0.ɵɵtemplate(2, CarouselComponent_ng_template_5_div_0_ng_template_2_Template, 4, 8, "ng-template", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var i_r4 = ɵngcc0.ɵɵnextContext().index;
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("width", ctx_r5.getCellWidth() + "px")("border-radius", ctx_r5.borderRadius + "px");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r5.getFile(i_r4) && ctx_r5.getFile(i_r4)["image"]);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r5.getFile(i_r4) && ctx_r5.getFile(i_r4)["type"] === "video");
} }
function CarouselComponent_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, CarouselComponent_ng_template_5_div_0_Template, 3, 6, "div", 8);
} if (rf & 2) {
    var i_r4 = ctx.index;
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", i_r4 < ctx_r1.cellLimit);
} }
function CarouselComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 17);
    ɵngcc0.ɵɵelementStart(1, "div", 18);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_div_6_Template_div_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.prev(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(2, "div", 19);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_div_6_Template_div_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r16 = ɵngcc0.ɵɵnextContext(); return ctx_r16.next(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("carousel-arrows-outside", ctx_r2.arrowsOutside)("carousel-dark-arrows", ctx_r2.arrowsTheme === "dark");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassProp("carousel-arrow-disabled", ctx_r2.isPrevArrowDisabled());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassProp("carousel-arrow-disabled", ctx_r2.isNextArrowDisabled());
} }
function OverlayContainerComponent_ng_template_0_Template(rf, ctx) { }
'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m) return m.call(o);
        if (o && typeof o.length === "number") return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }

    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    var Touches = /** @class */ (function () {
        function Touches(properties) {
            var _this = this;
            this.eventType = undefined;
            this.handlers = {};
            this.startX = 0;
            this.startY = 0;
            this.lastTap = 0;
            this.doubleTapMinTimeout = 300;
            this.tapMinTimeout = 200;
            this.touchstartTime = 0;
            this.i = 0;
            this.isMousedown = false;
            this.touchListeners = {
                "touchstart": "handleTouchstart",
                "touchmove": "handleTouchmove",
                "touchend": "handleTouchend"
            };
            this.mouseListeners = {
                "mousedown": "handleMousedown",
                "mousemove": "handleMousemove",
                "mouseup": "handleMouseup",
                "wheel": "handleWheel"
            };
            this.otherListeners = {
                "resize": "handleResize"
            };
            /*
             * Listeners
             */
            /* Touchstart */
            this.handleTouchstart = function (event) {
                _this.elementPosition = _this.getElementPosition();
                _this.touchstartTime = new Date().getTime();
                if (_this.eventType === undefined) {
                    _this.getTouchstartPosition(event);
                }
                _this.runHandler("touchstart", event);
            };
            /* Touchmove */
            this.handleTouchmove = function (event) {
                var touches = event.touches;
                // Pan
                if (_this.detectPan(touches)) {
                    _this.runHandler("pan", event);
                }
                // Pinch
                if (_this.detectPinch(event)) {
                    _this.runHandler("pinch", event);
                }
                // Linear swipe
                switch (_this.detectLinearSwipe(event)) {
                    case "horizontal-swipe":
                        event.swipeType = "horizontal-swipe";
                        _this.runHandler("horizontal-swipe", event);
                        break;
                    case "vertical-swipe":
                        event.swipeType = "vertical-swipe";
                        _this.runHandler("vertical-swipe", event);
                        break;
                }
                // Linear swipe
                if (_this.detectLinearSwipe(event) ||
                    _this.eventType === 'horizontal-swipe' ||
                    _this.eventType === 'vertical-swipe') {
                    _this.handleLinearSwipe(event);
                }
            };
            /* Touchend */
            this.handleTouchend = function (event) {
                var touches = event.touches;
                // Double Tap
                if (_this.detectDoubleTap()) {
                    _this.runHandler("double-tap", event);
                }
                // Tap
                _this.detectTap();
                _this.runHandler("touchend", event);
                _this.eventType = 'touchend';
                if (touches && touches.length === 0) {
                    _this.eventType = undefined;
                    _this.i = 0;
                }
            };
            /* Mousedown */
            this.handleMousedown = function (event) {
                _this.isMousedown = true;
                _this.elementPosition = _this.getElementPosition();
                _this.touchstartTime = new Date().getTime();
                if (_this.eventType === undefined) {
                    _this.getMousedownPosition(event);
                }
                _this.runHandler("mousedown", event);
            };
            /* Mousemove */
            this.handleMousemove = function (event) {
                //event.preventDefault();
                if (!_this.isMousedown) {
                    return;
                }
                // Pan
                _this.runHandler("pan", event);
                // Linear swipe
                switch (_this.detectLinearSwipe(event)) {
                    case "horizontal-swipe":
                        event.swipeType = "horizontal-swipe";
                        _this.runHandler("horizontal-swipe", event);
                        break;
                    case "vertical-swipe":
                        event.swipeType = "vertical-swipe";
                        _this.runHandler("vertical-swipe", event);
                        break;
                }
                // Linear swipe
                if (_this.detectLinearSwipe(event) ||
                    _this.eventType === 'horizontal-swipe' ||
                    _this.eventType === 'vertical-swipe') {
                    _this.handleLinearSwipe(event);
                }
            };
            /* Mouseup */
            this.handleMouseup = function (event) {
                // Tap
                _this.detectTap();
                _this.isMousedown = false;
                _this.runHandler("mouseup", event);
                _this.eventType = undefined;
                _this.i = 0;
            };
            /* Wheel */
            this.handleWheel = function (event) {
                _this.runHandler("wheel", event);
            };
            /* Resize */
            this.handleResize = function (event) {
                _this.runHandler("resize", event);
            };
            this.properties = properties;
            this.element = this.properties.element;
            this.elementPosition = this.getElementPosition();
            this.toggleEventListeners('addEventListener');
        }
        Touches.prototype.destroy = function () {
            this.toggleEventListeners('removeEventListener');
        };
        Touches.prototype.toggleEventListeners = function (action) {
            var listeners;
            if (this.properties.listeners === 'mouse and touch') {
                listeners = Object.assign(this.touchListeners, this.mouseListeners);
            }
            else {
                listeners = this.detectTouchScreen() ? this.touchListeners : this.mouseListeners;
            }
            if (this.properties.resize) {
                listeners = Object.assign(listeners, this.otherListeners);
            }
            for (var listener in listeners) {
                var handler = listeners[listener];
                // Window
                if (listener === "resize") {
                    if (action === 'addEventListener') {
                        window.addEventListener(listener, this[handler], false);
                    }
                    if (action === 'removeEventListener') {
                        window.removeEventListener(listener, this[handler], false);
                    }
                    // Document
                }
                else if (listener === 'mouseup' || listener === "mousemove") {
                    if (action === 'addEventListener') {
                        document.addEventListener(listener, this[handler], false);
                    }
                    if (action === 'removeEventListener') {
                        document.removeEventListener(listener, this[handler], false);
                    }
                    // Element
                }
                else {
                    if (action === 'addEventListener') {
                        this.element.addEventListener(listener, this[handler], false);
                    }
                    if (action === 'removeEventListener') {
                        this.element.removeEventListener(listener, this[handler], false);
                    }
                }
            }
        };
        Touches.prototype.handleLinearSwipe = function (event) {
            //event.preventDefault();
            this.i++;
            if (this.i > 3) {
                this.eventType = this.getLinearSwipeType(event);
            }
            if (this.eventType === 'horizontal-swipe') {
                this.runHandler('horizontal-swipe', event);
            }
            if (this.eventType === 'vertical-swipe') {
                this.runHandler('vertical-swipe', event);
            }
        };
        Touches.prototype.runHandler = function (eventName, response) {
            if (this.handlers[eventName]) {
                this.handlers[eventName](response);
            }
        };
        /*
         * Detection
         */
        Touches.prototype.detectPan = function (touches) {
            return touches.length === 1 && !this.eventType || this.eventType === 'pan';
        };
        Touches.prototype.detectDoubleTap = function () {
            var _this = this;
            if (this.eventType != undefined) {
                return;
            }
            var currentTime = new Date().getTime();
            var tapLength = currentTime - this.lastTap;
            clearTimeout(this.doubleTapTimeout);
            if (tapLength < this.doubleTapMinTimeout && tapLength > 0) {
                return true;
            }
            else {
                this.doubleTapTimeout = setTimeout(function () {
                    clearTimeout(_this.doubleTapTimeout);
                }, this.doubleTapMinTimeout);
            }
            this.lastTap = currentTime;
        };
        Touches.prototype.detectTap = function () {
            if (this.eventType != undefined) {
                return;
            }
            var currentTime = new Date().getTime();
            var tapLength = currentTime - this.touchstartTime;
            if (tapLength > 0) {
                if (tapLength < this.tapMinTimeout) {
                    this.runHandler("tap", event);
                }
                else {
                    this.runHandler("longtap", event);
                }
            }
        };
        Touches.prototype.detectPinch = function (event) {
            var touches = event.touches;
            return (touches.length === 2 && this.eventType === undefined) || this.eventType === 'pinch';
        };
        Touches.prototype.detectLinearSwipe = function (event) {
            var touches = event.touches;
            if (touches) {
                if (touches.length === 1 && !this.eventType || this.eventType === 'horizontal-swipe' || this.eventType === 'vertical-swipe') {
                    return this.getLinearSwipeType(event);
                }
            }
            else {
                if (!this.eventType || this.eventType === 'horizontal-swipe' || this.eventType === 'vertical-swipe') {
                    return this.getLinearSwipeType(event);
                }
            }
        };
        Touches.prototype.getLinearSwipeType = function (event) {
            if (this.eventType !== 'horizontal-swipe' && this.eventType !== 'vertical-swipe') {
                var movementX = Math.abs(this.moveLeft(0, event) - this.startX);
                var movementY = Math.abs(this.moveTop(0, event) - this.startY);
                if ((movementY * 3) > movementX) {
                    return 'vertical-swipe';
                }
                else {
                    return 'horizontal-swipe';
                }
            }
            else {
                return this.eventType;
            }
        };
        Touches.prototype.getElementPosition = function () {
            return this.element.getBoundingClientRect();
        };
        Touches.prototype.getTouchstartPosition = function (event) {
            this.startX = event.touches[0].clientX - this.elementPosition.left;
            this.startY = event.touches[0].clientY - this.elementPosition.top;
        };
        Touches.prototype.getMousedownPosition = function (event) {
            this.startX = event.clientX - this.elementPosition.left;
            this.startY = event.clientY - this.elementPosition.top;
        };
        Touches.prototype.moveLeft = function (index, event) {
            var touches = event.touches;
            if (touches) {
                return touches[index].clientX - this.elementPosition.left;
            }
            else {
                return event.clientX - this.elementPosition.left;
            }
        };
        Touches.prototype.moveTop = function (index, event) {
            var touches = event.touches;
            if (touches) {
                return touches[index].clientY - this.elementPosition.top;
            }
            else {
                return event.clientY - this.elementPosition.top;
            }
        };
        Touches.prototype.detectTouchScreen = function () {
            var prefixes = ' -webkit- -moz- -o- -ms- '.split(' ');
            var mq = function (query) {
                return window.matchMedia(query).matches;
            };
            if (('ontouchstart' in window)) {
                return true;
            }
            // include the 'heartz' as a way to have a non matching MQ to help terminate the join
            // https://git.io/vznFH
            var query = ['(', prefixes.join('touch-enabled),('), 'heartz', ')'].join('');
            return mq(query);
        };
        /* Public properties and methods */
        Touches.prototype.on = function (event, handler) {
            if (event) {
                this.handlers[event] = handler;
            }
        };
        return Touches;
    }());

    var Carousel = /** @class */ (function () {
        function Carousel(properties) {
            var _this = this;
            this.initialPositionX = 0;
            this.newContainerPositionIndex = 0; // The index of the new position relative to the active index, for example -1 or +1
            this.preliminarySlideCounter = 0;
            this.slideCounter = 0; // Scrolling Index (counter)
            this.previousSlideCounter = 0;
            this.isContentImages = true;
            this.isLazyLoad = true;
            this.isContainerLocked = false;
            this.alignCells = "left";
            this.initialContainerPosition = 0;
            this.handleTouchstart = function (event) {
                _this.isTouchstart = true;
                if (_this.isSlideInProgress) {
                    return;
                }
                var touches = event.touches;
                if (touches) {
                    _this.startX = touches[0].clientX - _this.getCarouselElementPosition()['left'];
                }
                else {
                    _this.startX = event.clientX - _this.getCarouselElementPosition()['left'];
                }
                _this.containerInitialPositionX = _this.getElementPosition()['left'] - _this.getCarouselElementPosition()['left'];
                _this.isMoveInProgress = true;
            };
            this.handleHorizontalSwipe = function (event) {
                if (_this.isSlideInProgress) {
                    return;
                }
                var touches = event.touches;
                if (touches) {
                    _this.moveX = touches[0].clientX - _this.getCarouselElementPosition()['left'];
                }
                else {
                    _this.moveX = event.clientX - _this.getCarouselElementPosition()['left'];
                }
                _this.distanceAbs = _this.getDistanceAbs();
                _this.direction = _this.getDirection();
                _this.moveContainer();
            };
            this.handleTouchend = function (event) {
                if (_this.isSlideInProgress || !_this.isTouchstart) {
                    _this.isTouchstart = false;
                    return;
                }
                _this.isMoveInProgress = false;
                if (_this.detectSlide()) {
                    _this.handleSlide();
                }
                else {
                    _this.newContainerPositionIndex = 0;
                    _this.alignContainer();
                }
                _this.startX = _this.moveX = _this.distanceAbs = undefined;
                _this.isTouchstart = false;
            };
            this.element = properties.element;
            this.properties = properties;
            this.cells = this.element.children;
            this.visibleWidth = this.properties.visibleWidth || this.element.parentElement.clientWidth;
            this.setContainerWidth();
            this.alignContainer(0);
        }
        Object.defineProperty(Carousel.prototype, "isFirstCell", {
            get: function () {
                return this.slideCounter === 0;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "lastCellIndex", {
            get: function () {
                return this.images.length ? (this.images.length - 1) : (this.cells.length - 1);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "overflowCellsLimit", {
            get: function () {
                return this.properties.overflowCellsLimit;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "cellLimit", {
            get: function () {
                return this.visibleCellsCount + this.overflowCellsLimit * 2;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "images", {
            get: function () {
                return this.properties.images;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "margin", {
            get: function () {
                return this.properties.margin;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "minSwipeDistance", {
            get: function () {
                return this.properties.minSwipeDistance;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "transitionDuration", {
            get: function () {
                return this.properties.transitionDuration;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "transitionTimingFunction", {
            get: function () {
                return this.properties.transitionTimingFunction;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "fullCellWidth", {
            get: function () {
                return this.properties.cellWidth + this.margin;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "visibleCellsCount", {
            get: function () {
                return Math.ceil(this.visibleWidth / this.fullCellWidth);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "lapCounter", {
            get: function () {
                return Math.floor(this.slideCounter / this.totalContainerCellsCount);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(Carousel.prototype, "visibleCellsOverflowContainer", {
            get: function () {
                return (this.visibleCellsCount * this.fullCellWidth - this.margin) > this.visibleWidth;
            },
            enumerable: true,
            configurable: true
        });
        /* Move */
        Carousel.prototype.moveContainer = function () {
            var positionX = this.getMovePositionX();
            this.transformPositionX(positionX, 0);
        };
        Carousel.prototype.getMovePositionX = function () {
            var distance = this.getDistance();
            return this.containerInitialPositionX - distance;
        };
        /* Align */
        Carousel.prototype.alignContainer = function (duration) {
            if (duration === void 0) { duration = this.transitionDuration; }
            var positionX = this.getContainerPosition();
            this.transformPositionX(positionX, duration);
            this.setInitialContainerPosition(positionX);
        };
        Carousel.prototype.getContainerPosition = function () {
            var correction = this.getContainerPositionCorrection();
            this.isContainerPositionCorrection = correction != 0;
            return (this.initialContainerPosition + this.newContainerPositionIndex * this.fullCellWidth) + correction;
        };
        Carousel.prototype.setInitialContainerPosition = function (position) {
            var correction = this.getContainerPositionCorrection();
            this.initialContainerPosition = position - correction;
        };
        Carousel.prototype.getContainerPositionCorrection = function () {
            var correction = 0;
            if ((this.images.length - this.preliminarySlideCounter) < this.visibleCellsCount || this.isSlideLengthLimited) {
                if (this.visibleWidth < this.totalContainerCellsCount * this.fullCellWidth) {
                    correction = -(this.visibleCellsCount * this.fullCellWidth - this.visibleWidth - this.margin);
                }
                if (correction >= -this.margin) {
                    correction = 0;
                }
            }
            return correction;
        };
        /* Quickly center */
        Carousel.prototype.quicklyPositionContainer = function () {
            var correction = this.getContainerPositionCorrection();
            var initialPosition = this.getNewContainerPosition() + correction;
            this.transformPositionX(initialPosition, 0);
            this.setInitialContainerPosition(initialPosition);
        };
        Carousel.prototype.getNewContainerPosition = function () {
            if (this.slideCounter > this.overflowCellsLimit) {
                if (this.alignCells === "left") {
                    this.lineUpCells();
                    return -(this.overflowCellsLimit * this.fullCellWidth);
                }
            }
            if (this.slideCounter <= this.overflowCellsLimit) {
                if (this.previousSlideCounter > this.overflowCellsLimit) {
                    this.lineUpCells();
                }
                return -(this.slideCounter * this.fullCellWidth);
            }
        };
        Carousel.prototype.setContainerWidth = function () {
            var containerWidth = this.getContainerWidth();
            this.element.style.width = containerWidth + "px";
        };
        Carousel.prototype.getContainerWidth = function () {
            this.totalContainerCellsCount = this.visibleCellsCount + this.overflowCellsLimit * 2;
            var containerWidth = this.totalContainerCellsCount * this.fullCellWidth;
            var totalImageWidth = this.images.length * this.fullCellWidth;
            if (totalImageWidth < containerWidth) {
                containerWidth = totalImageWidth;
            }
            return containerWidth;
        };
        Carousel.prototype.getFile = function (cellIndex) {
            var imageIndex = this.getFileIndex(cellIndex);
            var file = this.images[imageIndex];
            if (file && !file.type) {
                file.type = 'image';
            }
            return {
                image: this.images[imageIndex],
                imageIndex: imageIndex
            };
        };
        Carousel.prototype.getFileIndex = function (cellIndex) {
            var positionIndex = this.getPositionIndex(cellIndex);
            var numberLeftCells = (this.totalContainerCellsCount - 1) / 2;
            var imageIndex;
            if (!this.isContainerLocked) {
                if (this.slideCounter <= this.overflowCellsLimit) {
                    return cellIndex;
                }
                else {
                    var cellLimitOverflow = this.slideCounter - this.overflowCellsLimit;
                    imageIndex = positionIndex + cellLimitOverflow;
                    return imageIndex;
                }
            }
            if (this.alignCells === "left") {
                if (this.slideCounter > this.overflowCellsLimit) {
                    var cellLimitOverflow = this.slideCounter - this.overflowCellsLimit;
                    imageIndex = positionIndex + cellLimitOverflow;
                }
                else {
                    imageIndex = cellIndex;
                }
            }
            if (imageIndex > this.lastCellIndex) {
                return false;
            }
            return imageIndex;
        };
        Carousel.prototype.transformPositionX = function (value, duration) {
            if (duration === void 0) { duration = this.transitionDuration; }
            this.element.style.transition = 'transform ' + duration + 'ms ' + this.transitionTimingFunction;
            this.element.style.transform = 'translateX(' + value + 'px)';
        };
        Carousel.prototype.detectSlide = function () {
            return this.distanceAbs >= this.minSwipeDistance;
        };
        Carousel.prototype.next = function () {
            if (this.isSlideInProgress) {
                return;
            }
            this.direction = 'left';
            this.handleSlide(1);
        };
        Carousel.prototype.prev = function () {
            if (this.isSlideInProgress) {
                return;
            }
            this.direction = 'right';
            this.handleSlide(1);
        };
        Carousel.prototype.select = function (index) {
            this.slideCounter = index;
            this.quicklyPositionContainer();
        };
        Carousel.prototype.handleSlide = function (slideLength) {
            if (slideLength === void 0) { slideLength = undefined; }
            this.slideLength = this.getSlideLength();
            this.slideLength = slideLength ? slideLength : this.limitSlideLength(this.slideLength);
            if (this.direction === 'left' && !this.isSlideInProgress) {
                this.preliminarySlideCounter = this.slideCounter + this.slideLength;
                if (!this.detectLastSlide(this.slideCounter + this.slideLength)) {
                    this.newContainerPositionIndex = this.newContainerPositionIndex - this.slideLength;
                    this.isSlideInProgress = true;
                    if (this.isLazyLoad) {
                        this.isContainerLocked = this.preliminarySlideCounter > this.overflowCellsLimit;
                        if (this.detectContainerUnlock()) {
                            this.isContainerLocked = false;
                        }
                    }
                }
            }
            if (this.direction === 'right' && !this.isSlideInProgress) {
                if (this.slideCounter - this.slideLength < 0) {
                    this.slideLength = this.slideCounter;
                }
                this.preliminarySlideCounter = this.slideCounter - this.slideLength;
                if (!this.isFirstCell) {
                    this.newContainerPositionIndex = this.newContainerPositionIndex + this.slideLength;
                    this.isSlideInProgress = true;
                    if (this.isLazyLoad) {
                        if (this.preliminarySlideCounter > this.overflowCellsLimit) {
                            this.isContainerLocked = true;
                        }
                        else {
                            this.isContainerLocked = false;
                        }
                        if (this.detectContainerUnlock()) {
                            this.isContainerLocked = false;
                        }
                    }
                }
            }
            this.alignContainer();
        };
        Carousel.prototype.getSlideLength = function () {
            var correction = this.getContainerPositionCorrection();
            var length = Math.floor((this.distanceAbs + correction) / this.fullCellWidth);
            if (this.distanceAbs % this.fullCellWidth >= this.minSwipeDistance) {
                length++;
            }
            return length;
        };
        Carousel.prototype.limitSlideLength = function (slideLength) {
            if (slideLength > 1) {
                for (var i = 0; i < slideLength; i++) {
                    var newSlideCounter = this.slideCounter + (slideLength - i);
                    if (!this.detectLastSlide(newSlideCounter)) {
                        slideLength = slideLength - i;
                        this.isSlideLengthLimited = i > 0;
                        break;
                    }
                }
            }
            return slideLength;
        };
        Carousel.prototype.detectLastSlide = function (slideCounter) {
            return (this.images.length - slideCounter) < this.visibleCellsCount;
        };
        Carousel.prototype.isNextArrowDisabled = function () {
            if (this.visibleCellsOverflowContainer) {
                return this.detectLastSlide(this.slideCounter + 1) && this.isContainerPositionCorrection;
            }
            else {
                return this.detectLastSlide(this.slideCounter + 1);
            }
        };
        Carousel.prototype.isPrevArrowDisabled = function () {
            return this.slideCounter === 0;
        };
        Carousel.prototype.detectContainerUnlock = function () {
            return (this.images.length - this.preliminarySlideCounter) < (this.visibleCellsCount + this.overflowCellsLimit);
        };
        Carousel.prototype.handleSlideEnd = function () {
            if (this.isSlideInProgress) {
                this.transformSlideEnd();
                this.isSlideInProgress = false;
                this.newContainerPositionIndex = 0;
                this.isSlideLengthLimited = undefined;
            }
        };
        Carousel.prototype.transformSlideEnd = function () {
            if (this.isLazyLoad) {
                this.setSlideCounter();
                this.quicklyPositionContainer();
            }
            this.previousSlideCounter = this.slideCounter;
        };
        Carousel.prototype.setSlideCounter = function () {
            if (this.direction === 'left') {
                this.slideCounter = this.slideCounter + this.slideLength;
            }
            if (this.direction === 'right') {
                this.slideCounter = this.slideCounter - this.slideLength;
            }
            this.direction = undefined;
            this.slideLength = 0;
        };
        Carousel.prototype.resetTransition = function () {
            this.element.style.transition = '';
        };
        Carousel.prototype.getElementPosition = function () {
            return this.element.getBoundingClientRect();
        };
        Carousel.prototype.getCarouselElementPosition = function () {
            return this.properties.container.getBoundingClientRect();
        };
        Carousel.prototype.getDistance = function () {
            return this.startX - this.moveX;
        };
        Carousel.prototype.getDistanceAbs = function () {
            return Math.abs(this.startX - this.moveX);
        };
        Carousel.prototype.getDirection = function () {
            var direction = Math.sign(this.startX - this.moveX);
            if (direction === -1) {
                return 'right';
            }
            if (direction === 1) {
                return 'left';
            }
        };
        Carousel.prototype.lineUpCells = function () {
            var cells = this.element.children;
            for (var i = 0; i < cells.length; i++) {
                var cell = cells[i];
                var positionX = this.getCellPositionX(i);
                cell.style.transform = 'translateX(' + positionX + 'px)';
            }
            ;
        };
        Carousel.prototype.getCellPositionX = function (index) {
            var positionIndex = this.getPositionIndex(index);
            return positionIndex * this.fullCellWidth;
        };
        Carousel.prototype.getPositionIndex = function (cellIndex) {
            var slideCounter = this.slideCounter - this.overflowCellsLimit;
            var positionIndex;
            if (slideCounter > this.totalContainerCellsCount) {
                slideCounter = slideCounter % this.totalContainerCellsCount;
            }
            if (slideCounter < 0) {
                return cellIndex;
            }
            else {
                positionIndex = cellIndex - slideCounter;
                if (positionIndex < 0) {
                    positionIndex = this.totalContainerCellsCount + positionIndex;
                }
            }
            return positionIndex;
        };
        Object.defineProperty(Carousel.prototype, "containerOverflowRightCount", {
            get: function () {
                var totalOverflowCellsCount = this.images.length - this.totalContainerCellsCount;
                var overflowRight = 0;
                if (totalOverflowCellsCount > 0) {
                    overflowRight = totalOverflowCellsCount - (this.previousSlideCounter - this.overflowCellsLimit);
                    if (overflowRight > 0) {
                        return overflowRight;
                    }
                }
                return overflowRight;
            },
            enumerable: true,
            configurable: true
        });
        Carousel.prototype.getCenterPositionIndex = function () {
            return (this.totalContainerCellsCount - 1) / 2;
        };
        return Carousel;
    }());

    var EventService = /** @class */ (function () {
        function EventService() {
            this.emitter = new rxjs.ReplaySubject(1);
        }
        EventService.prototype.emitChangeEvent = function (data) {
            this.emitter.next(data);
        };
EventService.ɵfac = function EventService_Factory(t) { return new (t || EventService)(); };
EventService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: EventService, factory: function (t) { return EventService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EventService, [{
        type: core.Injectable
    }], function () { return []; }, null); })();
        return EventService;
    }());

    var CarouselComponent = /** @class */ (function () {
        function CarouselComponent(elementRef, ref, overlayEventService) {
            var _this = this;
            this.elementRef = elementRef;
            this.ref = ref;
            this.overlayEventService = overlayEventService;
            this.minTimeout = 30;
            this._cellWidth = 200;
            this._transitionDuration = 200;
            this.events = new core.EventEmitter();
            this.height = 200;
            this.margin = 10;
            this.objectFit = 'cover';
            this.minSwipeDistance = 50;
            this.transitionTimingFunction = 'ease';
            this.counterSeparator = " / ";
            this.overflowCellsLimit = 3;
            this.listeners = 'mouse and touch';
            this.arrows = true;
            this.arrowsTheme = 'light';
            this.hostClassCarousel = true;
            /* Touchstart */
            this.handleTouchstart = function (event) {
                event.preventDefault();
                _this.carousel.handleTouchstart(event);
                _this.isMoving = true;
                _this.events.emit({
                    type: 'touchstart',
                    event: event
                });
            };
            /* Touchmove */
            this.handleHorizontalSwipe = function (event) {
                event.preventDefault();
                _this.carousel.handleHorizontalSwipe(event);
                _this.events.emit({
                    type: 'swipe',
                    event: event
                });
            };
            /* Touchend */
            this.handleTouchend = function (event) {
                var touches = event.touches;
                _this.carousel.handleTouchend(event);
                _this.isMoving = false;
                _this.events.emit({
                    type: 'touchend',
                    event: event
                });
            };
            /* Tap */
            this.handleTap = function (event) {
                var i = _this.carousel.slideCounter;
                var cellIndex = _this.carousel.currentCellIndex;
                var fileIndex = _this.carousel.getFileIndex(i);
                var file = _this.carousel.getFile(cellIndex);
                _this.events.emit({
                    type: 'click',
                    file: file,
                    index: fileIndex
                });
            };
        }
        Object.defineProperty(CarouselComponent.prototype, "isLandscape", {
            get: function () {
                return window.innerWidth > window.innerHeight;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "isSafari", {
            get: function () {
                var ua = navigator.userAgent.toLowerCase();
                if (ua.indexOf('safari') !== -1) {
                    return !(ua.indexOf('chrome') > -1);
                }
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "counter", {
            get: function () {
                return this.carousel.slideCounter + 1 + this.counterSeparator + this.images.length;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "images", {
            get: function () {
                return this._images;
            },
            set: function (images) {
                this._images = images;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "isNgContent", {
            get: function () {
                return this.elementRef.nativeElement.querySelector('.carousel-content-wrapper').children.length > 0;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "transitionDuration", {
            get: function () {
                if (this._transitionDuration === 0) {
                    return 1;
                }
                else {
                    return this._transitionDuration;
                }
            },
            set: function (value) {
                this._transitionDuration = value;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "cellWidth", {
            set: function (value) {
                if (value) {
                    this._cellWidth = value;
                }
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "isCounter", {
            get: function () {
                return this._isCounter && this.images.length > 1;
            },
            set: function (value) {
                if (value) {
                    this._isCounter = value;
                }
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "overlayProperties", {
            set: function (properties) {
                this._properties = properties;
                this._images = this._properties.metadata.images;
                this._cellWidth = this._properties.metadata.width;
                this.height = this._properties.metadata.height;
                this.objectFit = this._properties.metadata.objectFit;
                this.margin = this._properties.metadata.margin;
                this.transitionDuration = this._properties.metadata.transitionDuration;
                this.overflowCellsLimit = this._properties.metadata.overflowCellsLimit;
                this.minSwipeDistance = this._properties.metadata.minSwipeDistance;
                this.transitionTimingFunction = this._properties.metadata.transitionTimingFunction;
                this._isCounter = this._properties.metadata.counter;
                this.counterSeparator = this._properties.metadata.counterSeparator;
                this.arrows = this._properties.metadata.arrows;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "properties", {
            get: function () {
                return this._properties;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(CarouselComponent.prototype, "cellLimit", {
            get: function () {
                if (this.carousel) {
                    return this.carousel.cellLimit;
                }
            },
            enumerable: true,
            configurable: true
        });
        CarouselComponent.prototype.onWindowResize = function (event) {
            this.landscapeMode = this.isLandscape;
            this.ref.detectChanges();
            this.initCarousel();
            this.carousel.lineUpCells();
        };
        CarouselComponent.prototype.ngOnInit = function () {
            this.touches = new Touches({
                element: this.elementRef.nativeElement.querySelector('.carousel-cells'),
                listeners: this.listeners
            });
            this.touches.on('touchstart', this.handleTouchstart);
            this.touches.on('horizontal-swipe', this.handleHorizontalSwipe);
            this.touches.on('touchend', this.handleTouchend);
            this.touches.on('mousedown', this.handleTouchstart);
            this.touches.on('mouseup', this.handleTouchend);
            this.touches.on('tap', this.handleTap);
            this.initCarousel();
            this.setDimensions();
            this.setInitialIndex();
        };
        CarouselComponent.prototype.ngAfterViewInit = function () {
            this.carousel.lineUpCells();
        };
        CarouselComponent.prototype.ngOnChanges = function (changes) {
            if (changes.width || changes.height) {
                this.setDimensions();
                this.initCarousel();
                this.carousel.lineUpCells();
            }
        };
        CarouselComponent.prototype.ngOnDestroy = function () {
            this.touches.destroy();
        };
        CarouselComponent.prototype.initCarousel = function () {
            this.carousel = new Carousel({
                element: this.elementRef.nativeElement.querySelector('.carousel-cells'),
                container: this.elementRef.nativeElement,
                images: this.images,
                cellWidth: this.getCellWidth(),
                overflowCellsLimit: this.overflowCellsLimit,
                visibleWidth: this.width,
                margin: this.margin,
                minSwipeDistance: this.minSwipeDistance,
                transitionDuration: this.transitionDuration,
                transitionTimingFunction: this.transitionTimingFunction,
                videoProperties: this.videoProperties
            });
        };
        CarouselComponent.prototype.setDimensions = function () {
            this.hostStyleHeight = this.height + 'px';
            this.hostStyleWidth = this.width + 'px';
        };
        CarouselComponent.prototype.getFile = function (index) {
            return this.carousel.getFile(index);
        };
        CarouselComponent.prototype.handleTransitionendCellContainer = function (event) {
            this.carousel.handleSlideEnd();
        };
        CarouselComponent.prototype.toggleVideo = function (video) {
            event.preventDefault();
            if (this.videoProperties.noPlay) {
                return;
            }
            if (video.paused) {
                video.play();
                this.isVideoPlaying = true;
            }
            else {
                video.pause();
                this.isVideoPlaying = false;
            }
            this.ref.detectChanges();
        };
        CarouselComponent.prototype.getCurrentIndex = function () {
            return this.carousel.slideCounter;
        };
        CarouselComponent.prototype.getCellWidth = function () {
            if (this._cellWidth === '100%') {
                return this.elementRef.nativeElement.clientWidth;
            }
            else {
                return this._cellWidth;
            }
        };
        CarouselComponent.prototype.next = function () {
            this.carousel.next(1);
        };
        CarouselComponent.prototype.prev = function () {
            this.carousel.prev(1);
        };
        CarouselComponent.prototype.select = function (index) {
            this.carousel.select(index);
        };
        CarouselComponent.prototype.isNextArrowDisabled = function () {
            return this.carousel.isNextArrowDisabled();
        };
        CarouselComponent.prototype.isPrevArrowDisabled = function () {
            return this.carousel.isPrevArrowDisabled();
        };
        CarouselComponent.prototype.close = function () {
            this.overlayEventService.emitChangeEvent({
                type: 'Hide'
            });
        };
        CarouselComponent.prototype.setInitialIndex = function () {
            if (this._properties.metadata.index != undefined) {
                this.select(this._properties.metadata.index);
            }
        };
        CarouselComponent.ctorParameters = function () { return [
            { type: core.ElementRef },
            { type: core.ChangeDetectorRef },
            { type: EventService }
        ]; };
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "images", null);
        __decorate([
            core.Output()
        ], CarouselComponent.prototype, "events", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "height", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "width", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "borderRadius", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "margin", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "objectFit", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "minSwipeDistance", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "transitionDuration", null);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "transitionTimingFunction", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "videoProperties", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "counterSeparator", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "overflowCellsLimit", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "listeners", void 0);
        __decorate([
            core.Input('cellWidth')
        ], CarouselComponent.prototype, "cellWidth", null);
        __decorate([
            core.Input('counter')
        ], CarouselComponent.prototype, "isCounter", null);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "arrows", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "arrowsOutside", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "arrowsTheme", void 0);
        __decorate([
            core.Input()
        ], CarouselComponent.prototype, "overlayProperties", null);
        __decorate([
            core.HostBinding('class.carousel')
        ], CarouselComponent.prototype, "hostClassCarousel", void 0);
        __decorate([
            core.HostBinding('style.height')
        ], CarouselComponent.prototype, "hostStyleHeight", void 0);
        __decorate([
            core.HostBinding('style.width')
        ], CarouselComponent.prototype, "hostStyleWidth", void 0);
        __decorate([
            core.HostListener('window:resize', ['$event'])
        ], CarouselComponent.prototype, "onWindowResize", null);
CarouselComponent.ɵfac = function CarouselComponent_Factory(t) { return new (t || CarouselComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(EventService)); };
CarouselComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselComponent, selectors: [["carousel"], ["", "carousel", ""]], hostVars: 6, hostBindings: function CarouselComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("resize", function CarouselComponent_resize_HostBindingHandler($event) { return ctx.onWindowResize($event); }, false, ɵngcc0.ɵɵresolveWindow);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("height", ctx.hostStyleHeight)("width", ctx.hostStyleWidth);
        ɵngcc0.ɵɵclassProp("carousel", ctx.hostClassCarousel);
    } }, inputs: { height: "height", margin: "margin", objectFit: "objectFit", minSwipeDistance: "minSwipeDistance", transitionTimingFunction: "transitionTimingFunction", counterSeparator: "counterSeparator", overflowCellsLimit: "overflowCellsLimit", listeners: "listeners", arrows: "arrows", arrowsTheme: "arrowsTheme", images: "images", transitionDuration: "transitionDuration", cellWidth: "cellWidth", isCounter: ["counter", "isCounter"], overlayProperties: "overlayProperties", width: "width", borderRadius: "borderRadius", videoProperties: "videoProperties", arrowsOutside: "arrowsOutside" }, outputs: { events: "events" }, exportAs: ["carousel"], features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 7, vars: 5, consts: [[1, "carousel-backdrop"], [1, "carousel-close", 3, "click"], ["class", "carousel-counter", 4, "ngIf"], [1, "carousel-container"], [1, "carousel-cells", 3, "transitionend"], ["ngFor", "", 3, "ngForOf"], ["class", "carousel-arrows", 3, "carousel-arrows-outside", "carousel-dark-arrows", 4, "ngIf"], [1, "carousel-counter"], ["class", "carousel-cell", 3, "width", "border-radius", 4, "ngIf"], [1, "carousel-cell"], ["draggable", "false", 3, "src", "object-fit", 4, "ngIf"], [3, "ngIf"], ["draggable", "false", 3, "src"], [1, "carousel-play", 3, "click"], [3, "loop", "muted", "controls"], ["video", ""], ["type", "video/mp4", 3, "src"], [1, "carousel-arrows"], [1, "carousel-arrow", "carousel-arrow-prev", 3, "click"], [1, "carousel-arrow", "carousel-arrow-next", 3, "click"]], template: function CarouselComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵlistener("click", function CarouselComponent_Template_div_click_1_listener() { return ctx.close(); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(2, CarouselComponent_div_2_Template, 2, 1, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵelementStart(4, "div", 4);
        ɵngcc0.ɵɵlistener("transitionend", function CarouselComponent_Template_div_transitionend_4_listener($event) { return ctx.handleTransitionendCellContainer($event); });
        ɵngcc0.ɵɵtemplate(5, CarouselComponent_ng_template_5_Template, 1, 1, "ng-template", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, CarouselComponent_div_6_Template, 3, 8, "div", 6);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isCounter);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassProp("carousel-moving", ctx.isMoving);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.images);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.arrows);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgForOf], styles: ["[_nghost-%COMP%]{position:relative;display:block;top:0;left:0;width:100%;height:100%;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;z-index:10000;transform-origin:top left;box-sizing:border-box}[_nghost-%COMP%]   .carousel-container[_ngcontent-%COMP%]{overflow:hidden;width:100%;height:100%;cursor:-webkit-grab;cursor:grab}[_nghost-%COMP%]   .carousel-container.carousel-moving[_ngcontent-%COMP%]{cursor:-webkit-grabbing;cursor:grabbing}[_nghost-%COMP%]   .carousel-counter[_ngcontent-%COMP%]{text-align:right;position:absolute;z-index:30;transition:opacity .2s;top:15px;left:10px;border-radius:13px;background-color:rgba(23,37,68,.3);font-size:13px;color:#fff;padding:5px 7px;line-height:initial}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]{transition:transform .2s;width:100%;height:100%;display:block}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell.swiper-prev-image{transform:translate3d(-100%,0,0)}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell.swiper-next-image{transform:translate3d(100%,0,0)}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell{width:100%;height:100%;text-align:center;position:absolute;display:flex;align-items:center;justify-content:center;overflow:hidden}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell img, [_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell video{width:100%;height:100%;position:relative;-o-object-fit:contain;object-fit:contain}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell img.swiper-hide{display:none}[_nghost-%COMP%]   .carousel-cells[_ngcontent-%COMP%]     .carousel-cell .carousel-play{position:absolute;top:0;left:0;bottom:0;right:0;z-index:1}[_nghost-%COMP%]   .carousel-arrow[_ngcontent-%COMP%]{width:40px;height:40px;background-color:#fff;background-repeat:no-repeat;background-size:31px;background-position:center;border-radius:100px;position:absolute;top:50%;margin-top:-20px;z-index:10;cursor:pointer;box-shadow:0 0 5px rgba(0,0,0,.15)}[_nghost-%COMP%]   .carousel-arrow-prev[_ngcontent-%COMP%]{left:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTUuNDEgMTYuNTlMMTAuODMgMTJsNC41OC00LjU5TDE0IDZsLTYgNiA2IDYgMS40MS0xLjQxeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMFYweiIgZmlsbD0ibm9uZSIvPjwvc3ZnPg==)}[_nghost-%COMP%]   .carousel-arrow-next[_ngcontent-%COMP%]{right:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNOC41OSAxNi41OUwxMy4xNyAxMiA4LjU5IDcuNDEgMTAgNmw2IDYtNiA2LTEuNDEtMS40MXoiLz48cGF0aCBkPSJNMCAwaDI0djI0SDBWMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=)}[_nghost-%COMP%]   .carousel-arrows-outside[_ngcontent-%COMP%]   .carousel-arrow-prev[_ngcontent-%COMP%]{left:-60px}[_nghost-%COMP%]   .carousel-arrows-outside[_ngcontent-%COMP%]   .carousel-arrow-next[_ngcontent-%COMP%]{right:-60px}[_nghost-%COMP%]   .carousel-dark-arrows[_ngcontent-%COMP%]   .carousel-arrow[_ngcontent-%COMP%]{-webkit-filter:invert(1);filter:invert(1)}[_nghost-%COMP%]   .carousel-arrow-disabled[_ngcontent-%COMP%]{cursor:default;opacity:.5}[_nghost-%COMP%]   .carousel-close[_ngcontent-%COMP%]{position:fixed;top:10px;right:10px;width:40px;height:40px;z-index:100;cursor:pointer;background-repeat:no-repeat;background-size:31px;background-position:center;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTkgNi40MUwxNy41OSA1IDEyIDEwLjU5IDYuNDEgNSA1IDYuNDEgMTAuNTkgMTIgNSAxNy41OSA2LjQxIDE5IDEyIDEzLjQxIDE3LjU5IDE5IDE5IDE3LjU5IDEzLjQxIDEyeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=);-webkit-filter:invert(1);filter:invert(1)}[_nghost-%COMP%]   .carousel-backdrop[_ngcontent-%COMP%]{position:fixed;top:0;left:0;right:0;bottom:0;background-color:#000;opacity:.5}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CarouselComponent, [{
        type: core.Component,
        args: [{
                selector: 'carousel, [carousel]',
                exportAs: 'carousel',
                template: "<div class=\"carousel-backdrop\"></div>\n\n<div class=\"carousel-close\" (click)=\"close()\"></div>\n\n<div class=\"carousel-counter\" *ngIf=\"isCounter\">{{counter}}</div>\n\n<div class=\"carousel-container\" [class.carousel-moving]=\"isMoving\">\n\t<div class=\"carousel-cells\" (transitionend)=\"handleTransitionendCellContainer($event)\">\n\t\t<ng-template ngFor let-image [ngForOf]=\"images\" let-i=\"index\">\n\t\t\t<div class=\"carousel-cell\" \n\t\t\t\t*ngIf=\"i < cellLimit\" \n\t\t\t\t[style.width]=\"getCellWidth()+'px'\"\n\t\t\t\t[style.border-radius]=\"borderRadius+'px'\">\n\t\t\t\t<!-- Image -->\n\t\t\t\t<img \n\t\t\t\t\t*ngIf=\"getFile(i) && getFile(i)['image']\" \n\t\t\t\t\t[src]=\"getFile(i)['image']['path']\"\n\t\t\t\t\t[style.object-fit]=\"objectFit\"\n\t\t\t\t\tdraggable=\"false\" />\n\n\t\t\t\t<!-- Video -->\n\t\t\t\t<ng-template [ngIf]=\"getFile(i) && getFile(i)['type'] === 'video'\">\n\t\t\t\t\t<div class=\"carousel-play\" (click)=\"toggleVideo(video)\"></div>\n\n\t\t\t\t\t<video #video\n\t\t\t\t\t\t[loop]=\"videoProperties.loop\"\n\t\t\t\t\t\t[muted]=\"videoProperties.muted\"\n\t\t\t\t\t\t[controls]=\"isSafari\"\n\t\t\t\t\t\t[style.object-fit]=\"objectFit\"\n\t\t\t\t\t\t[style.border-radius]=\"borderRadius+'px'\">\n\t\t\t\t\t\t\n\t\t\t\t\t\t<source [src]=\"getFile(i)['path']\" type=\"video/mp4\">\n\t\t\t\t\t</video>\n\t\t\t\t</ng-template>\n\t\t\t</div>\n\t\t</ng-template>\n\t</div>\n</div>\n\n<div class=\"carousel-arrows\" \n\t[class.carousel-arrows-outside]=\"arrowsOutside\" \n\t[class.carousel-dark-arrows]=\"arrowsTheme === 'dark'\"\n\t*ngIf=\"arrows\">\n\t\n\t<div class=\"carousel-arrow carousel-arrow-prev\" [class.carousel-arrow-disabled]=\"isPrevArrowDisabled()\" (click)=\"prev()\"></div>\n\t<div class=\"carousel-arrow carousel-arrow-next\" [class.carousel-arrow-disabled]=\"isNextArrowDisabled()\" (click)=\"next()\"></div>\n</div>",
                styles: [":host{position:relative;display:block;top:0;left:0;width:100%;height:100%;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;z-index:10000;transform-origin:top left;box-sizing:border-box}:host .carousel-container{overflow:hidden;width:100%;height:100%;cursor:-webkit-grab;cursor:grab}:host .carousel-container.carousel-moving{cursor:-webkit-grabbing;cursor:grabbing}:host .carousel-counter{text-align:right;position:absolute;z-index:30;transition:opacity .2s;top:15px;left:10px;border-radius:13px;background-color:rgba(23,37,68,.3);font-size:13px;color:#fff;padding:5px 7px;line-height:initial}:host .carousel-cells{transition:transform .2s;width:100%;height:100%;display:block}:host .carousel-cells ::ng-deep .carousel-cell.swiper-prev-image{transform:translate3d(-100%,0,0)}:host .carousel-cells ::ng-deep .carousel-cell.swiper-next-image{transform:translate3d(100%,0,0)}:host .carousel-cells ::ng-deep .carousel-cell{width:100%;height:100%;text-align:center;position:absolute;display:flex;align-items:center;justify-content:center;overflow:hidden}:host .carousel-cells ::ng-deep .carousel-cell img,:host .carousel-cells ::ng-deep .carousel-cell video{width:100%;height:100%;position:relative;-o-object-fit:contain;object-fit:contain}:host .carousel-cells ::ng-deep .carousel-cell img.swiper-hide{display:none}:host .carousel-cells ::ng-deep .carousel-cell .carousel-play{position:absolute;top:0;left:0;bottom:0;right:0;z-index:1}:host .carousel-arrow{width:40px;height:40px;background-color:#fff;background-repeat:no-repeat;background-size:31px;background-position:center;border-radius:100px;position:absolute;top:50%;margin-top:-20px;z-index:10;cursor:pointer;box-shadow:0 0 5px rgba(0,0,0,.15)}:host .carousel-arrow-prev{left:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTUuNDEgMTYuNTlMMTAuODMgMTJsNC41OC00LjU5TDE0IDZsLTYgNiA2IDYgMS40MS0xLjQxeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMFYweiIgZmlsbD0ibm9uZSIvPjwvc3ZnPg==)}:host .carousel-arrow-next{right:10px;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNOC41OSAxNi41OUwxMy4xNyAxMiA4LjU5IDcuNDEgMTAgNmw2IDYtNiA2LTEuNDEtMS40MXoiLz48cGF0aCBkPSJNMCAwaDI0djI0SDBWMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=)}:host .carousel-arrows-outside .carousel-arrow-prev{left:-60px}:host .carousel-arrows-outside .carousel-arrow-next{right:-60px}:host .carousel-dark-arrows .carousel-arrow{-webkit-filter:invert(1);filter:invert(1)}:host .carousel-arrow-disabled{cursor:default;opacity:.5}:host .carousel-close{position:fixed;top:10px;right:10px;width:40px;height:40px;z-index:100;cursor:pointer;background-repeat:no-repeat;background-size:31px;background-position:center;background-image:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGhlaWdodD0iMjQiIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0Ij48cGF0aCBkPSJNMTkgNi40MUwxNy41OSA1IDEyIDEwLjU5IDYuNDEgNSA1IDYuNDEgMTAuNTkgMTIgNSAxNy41OSA2LjQxIDE5IDEyIDEzLjQxIDE3LjU5IDE5IDE5IDE3LjU5IDEzLjQxIDEyeiIvPjxwYXRoIGQ9Ik0wIDBoMjR2MjRIMHoiIGZpbGw9Im5vbmUiLz48L3N2Zz4=);-webkit-filter:invert(1);filter:invert(1)}:host .carousel-backdrop{position:fixed;top:0;left:0;right:0;bottom:0;background-color:#000;opacity:.5}"]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.ChangeDetectorRef }, { type: EventService }]; }, { events: [{
            type: core.Output
        }], height: [{
            type: core.Input
        }], margin: [{
            type: core.Input
        }], objectFit: [{
            type: core.Input
        }], minSwipeDistance: [{
            type: core.Input
        }], transitionTimingFunction: [{
            type: core.Input
        }], counterSeparator: [{
            type: core.Input
        }], overflowCellsLimit: [{
            type: core.Input
        }], listeners: [{
            type: core.Input
        }], arrows: [{
            type: core.Input
        }], arrowsTheme: [{
            type: core.Input
        }], hostClassCarousel: [{
            type: core.HostBinding,
            args: ['class.carousel']
        }], images: [{
            type: core.Input
        }], transitionDuration: [{
            type: core.Input
        }], cellWidth: [{
            type: core.Input,
            args: ['cellWidth']
        }], isCounter: [{
            type: core.Input,
            args: ['counter']
        }], overlayProperties: [{
            type: core.Input
        }], onWindowResize: [{
            type: core.HostListener,
            args: ['window:resize', ['$event']]
        }], hostStyleHeight: [{
            type: core.HostBinding,
            args: ['style.height']
        }], hostStyleWidth: [{
            type: core.HostBinding,
            args: ['style.width']
        }], width: [{
            type: core.Input
        }], borderRadius: [{
            type: core.Input
        }], videoProperties: [{
            type: core.Input
        }], arrowsOutside: [{
            type: core.Input
        }] }); })();
        return CarouselComponent;
    }());

    var AdDirective = /** @class */ (function () {
        function AdDirective(viewContainerRef) {
            this.viewContainerRef = viewContainerRef;
        }
        AdDirective.ctorParameters = function () { return [
            { type: core.ViewContainerRef }
        ]; };
AdDirective.ɵfac = function AdDirective_Factory(t) { return new (t || AdDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
AdDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: AdDirective, selectors: [["", "ad-host", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AdDirective, [{
        type: core.Directive,
        args: [{
                selector: '[ad-host]'
            }]
    }], function () { return [{ type: ɵngcc0.ViewContainerRef }]; }, null); })();
        return AdDirective;
    }());

    var OverlayContainerComponent = /** @class */ (function () {
        function OverlayContainerComponent(componentFactoryResolver, elementRef, eventService) {
            var _this = this;
            this.componentFactoryResolver = componentFactoryResolver;
            this.elementRef = elementRef;
            this.eventService = eventService;
            this.overlayBackdrop = true;
            this.minTimeout = 30;
            this._hostOpacity = 0;
            this.globalEventsSubscription = this.eventService.emitter.subscribe(function (event) {
                _this.handleGlobalEvents(event);
            });
        }
        Object.defineProperty(OverlayContainerComponent.prototype, "hostOpacity", {
            get: function () {
                return this.properties.fadeIn ? this._hostOpacity : 1;
            },
            set: function (value) {
                this._hostOpacity = value;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(OverlayContainerComponent.prototype, "hostTransition", {
            get: function () {
                var duration = this.properties.animationDuration + 'ms ';
                var esing = this.properties.animationTimingFunction;
                return 'opacity ' + duration + esing + ', transform ' + duration + esing;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(OverlayContainerComponent.prototype, "hostZIndex", {
            get: function () {
                return this.properties.zIndex;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(OverlayContainerComponent.prototype, "hostWidth", {
            get: function () {
                return this.properties.wrapperWidth;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(OverlayContainerComponent.prototype, "hostHeight", {
            get: function () {
                return this.properties.wrapperHeight;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(OverlayContainerComponent.prototype, "hostPosition", {
            get: function () {
                return this.properties.position;
            },
            enumerable: true,
            configurable: true
        });
        OverlayContainerComponent.prototype.onClick = function (event) {
            if (!this.elementRef.nativeElement.contains(event.target)) {
                if (this.hostShown) {
                    this.closeOverlay();
                }
            }
        };
        OverlayContainerComponent.prototype.transitionEnd = function (event) {
            if (event.target.tagName.toLowerCase() != "overlay") {
                return;
            }
            if (event.propertyName != "opacity") {
                return;
            }
            if (this.hostShown) {
                this.closeOverlayEnd();
                //this.stateEvents.emit("shown");
            }
            else {
                this.hostShown = true;
                //this.stateEvents.emit("hidden");
            }
        };
        Object.defineProperty(OverlayContainerComponent.prototype, "popoverClass", {
            get: function () {
                if (this.properties.metadata) {
                    return this.properties.metadata.popoverClass;
                }
            },
            enumerable: true,
            configurable: true
        });
        OverlayContainerComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.loadComponent();
            this.initialOverlay();
            setTimeout(function () {
                _this.isInitial = true;
            }, this.minTimeout); // Note: not good
        };
        OverlayContainerComponent.prototype.ngAfterViewInit = function () {
            var _this = this;
            setTimeout(function () {
                _this.showOverlayAnimation();
            }, this.minTimeout);
        };
        OverlayContainerComponent.prototype.ngOnDestroy = function () {
            if (this.globalEventsSubscription) {
                this.globalEventsSubscription.unsubscribe();
            }
        };
        OverlayContainerComponent.prototype.initialOverlay = function () {
            this.hostTransform = 'translateY(' + this.properties.animationTranslateY + ')';
        };
        OverlayContainerComponent.prototype.showOverlayAnimation = function () {
            this.hostOpacity = 1;
            this.hostTransform = 'translateY(0px)';
        };
        OverlayContainerComponent.prototype.loadComponent = function () {
            var adItem = this.properties;
            var componentFactory = this.componentFactoryResolver.resolveComponentFactory(adItem.mainComponent);
            var viewContainerRef = this.adHost.viewContainerRef;
            viewContainerRef.clear();
            var componentRef = viewContainerRef.createComponent(componentFactory);
            var componentRefElement = componentRef.location.nativeElement;
            // Set styles
            componentRefElement.style.width = this.properties.width;
            componentRefElement.style.height = this.properties.height;
            // Set classes
            if (typeof this.popoverClass === 'string') {
                componentRefElement.classList.add(this.popoverClass);
            }
            if (typeof this.popoverClass === 'object') {
                this.popoverClass.forEach(function (className) {
                    componentRefElement.classList.add(className);
                });
            }
            // Pass properties
            componentRef.instance.overlayProperties = this.properties;
        };
        OverlayContainerComponent.prototype.closeOverlay = function () {
            this.hostOpacity = 0;
            this.hostTransform = 'translateY(' + this.properties.animationTranslateY + ')';
        };
        OverlayContainerComponent.prototype.closeOverlayEnd = function () {
            this.hostShown = false;
            this.eventService.emitChangeEvent({
                type: '[Overlay] Hide'
            });
        };
        OverlayContainerComponent.prototype.handleGlobalEvents = function (event) {
            if (event.type === 'Hide') {
                this.closeOverlay();
            }
        };
        OverlayContainerComponent.ctorParameters = function () { return [
            { type: core.ComponentFactoryResolver },
            { type: core.ElementRef },
            { type: EventService }
        ]; };
        __decorate([
            core.ViewChild(AdDirective, { static: true })
        ], OverlayContainerComponent.prototype, "adHost", void 0);
        __decorate([
            core.Input()
        ], OverlayContainerComponent.prototype, "properties", void 0);
        __decorate([
            core.HostBinding('style.opacity')
        ], OverlayContainerComponent.prototype, "hostOpacity", null);
        __decorate([
            core.HostBinding('style.transition')
        ], OverlayContainerComponent.prototype, "hostTransition", null);
        __decorate([
            core.HostBinding('style.transform')
        ], OverlayContainerComponent.prototype, "hostTransform", void 0);
        __decorate([
            core.HostBinding('style.z-index')
        ], OverlayContainerComponent.prototype, "hostZIndex", null);
        __decorate([
            core.HostBinding('style.width')
        ], OverlayContainerComponent.prototype, "hostWidth", null);
        __decorate([
            core.HostBinding('style.height')
        ], OverlayContainerComponent.prototype, "hostHeight", null);
        __decorate([
            core.HostBinding('style.position')
        ], OverlayContainerComponent.prototype, "hostPosition", null);
        __decorate([
            core.HostBinding('class.overlay-shown')
        ], OverlayContainerComponent.prototype, "hostShown", void 0);
        __decorate([
            core.HostListener('document:click', ['$event'])
        ], OverlayContainerComponent.prototype, "onClick", null);
        __decorate([
            core.HostListener('transitionend', ['$event'])
        ], OverlayContainerComponent.prototype, "transitionEnd", null);
OverlayContainerComponent.ɵfac = function OverlayContainerComponent_Factory(t) { return new (t || OverlayContainerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(EventService)); };
OverlayContainerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: OverlayContainerComponent, selectors: [["overlay"]], viewQuery: function OverlayContainerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵstaticViewQuery(AdDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.adHost = _t.first);
    } }, hostVars: 16, hostBindings: function OverlayContainerComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function OverlayContainerComponent_click_HostBindingHandler($event) { return ctx.onClick($event); }, false, ɵngcc0.ɵɵresolveDocument)("transitionend", function OverlayContainerComponent_transitionend_HostBindingHandler($event) { return ctx.transitionEnd($event); });
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("opacity", ctx.hostOpacity)("transition", ctx.hostTransition)("z-index", ctx.hostZIndex)("width", ctx.hostWidth)("height", ctx.hostHeight)("position", ctx.hostPosition)("transform", ctx.hostTransform);
        ɵngcc0.ɵɵclassProp("overlay-shown", ctx.hostShown);
    } }, inputs: { properties: "properties" }, decls: 1, vars: 0, consts: [["ad-host", ""]], template: function OverlayContainerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, OverlayContainerComponent_ng_template_0_Template, 0, 0, "ng-template", 0);
    } }, directives: [AdDirective], styles: ["[_nghost-%COMP%]{position:absolute;top:0;left:0}[_nghost-%COMP%]   .overlay-backdrop[_ngcontent-%COMP%]{position:fixed;top:0;left:0;right:0;bottom:0;background-color:#000;opacity:0;z-index:1000;pointer-events:auto}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OverlayContainerComponent, [{
        type: core.Component,
        args: [{
                selector: 'overlay',
                template: "<!--\n<div class=\"overlay-backdrop\" \n\t*ngIf=\"properties.overlayBackdrop\" \n\t(click)=\"onBackdropClick()\" \n\t[ngClass]=\"properties.backdropClass\"></div>\n-->\n<!--\n<div class=\"overlay-container\"\n\t[ngStyle]=\"containerStyles\" \n\t[ngClass]=\"properties.overlayClass\">\n\n\t<ng-template ad-host></ng-template>\n</div>\n-->\n\n<ng-template ad-host></ng-template>",
                styles: [":host{position:absolute;top:0;left:0}:host .overlay-backdrop{position:fixed;top:0;left:0;right:0;bottom:0;background-color:#000;opacity:0;z-index:1000;pointer-events:auto}"]
            }]
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }, { type: ɵngcc0.ElementRef }, { type: EventService }]; }, { hostOpacity: [{
            type: core.HostBinding,
            args: ['style.opacity']
        }], hostTransition: [{
            type: core.HostBinding,
            args: ['style.transition']
        }], hostZIndex: [{
            type: core.HostBinding,
            args: ['style.z-index']
        }], hostWidth: [{
            type: core.HostBinding,
            args: ['style.width']
        }], hostHeight: [{
            type: core.HostBinding,
            args: ['style.height']
        }], hostPosition: [{
            type: core.HostBinding,
            args: ['style.position']
        }], onClick: [{
            type: core.HostListener,
            args: ['document:click', ['$event']]
        }], transitionEnd: [{
            type: core.HostListener,
            args: ['transitionend', ['$event']]
        }], hostShown: [{
            type: core.HostBinding,
            args: ['class.overlay-shown']
        }], hostTransform: [{
            type: core.HostBinding,
            args: ['style.transform']
        }], adHost: [{
            type: core.ViewChild,
            args: [AdDirective, { static: true }]
        }], properties: [{
            type: core.Input
        }] }); })();
        return OverlayContainerComponent;
    }());

    var defaultProperties = {
        animationDuration: 100,
        animationTimingFunction: 'ease-in-out',
        position: 'absolute',
        zIndex: 1000,
        fadeIn: true
    };

    var Overlay = /** @class */ (function () {
        function Overlay(componentFactoryResolver, appRef, injector, eventService) {
            var _this = this;
            this.componentFactoryResolver = componentFactoryResolver;
            this.appRef = appRef;
            this.injector = injector;
            this.eventService = eventService;
            this.componentRefs = {};
            this._properties = {};
            this.globalEventsSubscription = this.eventService.emitter.subscribe(function (event) {
                _this.handleGlobalEvents(event);
            });
        }
        Overlay.prototype.appendComponentToBody = function (properties, component) {
            if (component === void 0) { component = OverlayContainerComponent; }
            //if (this.componentRefs[properties.id]){
            if (this.componentRefs[0]) {
                return;
            }
            var componentRef = this.componentFactoryResolver
                .resolveComponentFactory(component)
                .create(this.injector);
            //this.componentRefs[properties.id] = componentRef;
            this.componentRefs[0] = componentRef;
            componentRef.instance.properties = properties;
            this.appRef.attachView(componentRef.hostView);
            var domElem = componentRef.hostView.rootNodes[0];
            // Add to body
            document.body.appendChild(domElem);
        };
        /*
        appendComponentToTag(element: any, tagname: string = ''):void {
            if (tagname){
                document.getElementsByTagName(tagname)[0].appendChild(element);
            } else {
                document.body.appendChild(element);
            }
        }
        */
        Overlay.prototype.load = function (properties) {
            properties = this.applyPropertieDefaults(defaultProperties, properties);
            this.appendComponentToBody(properties);
        };
        Overlay.prototype.close = function (prop) {
            if (prop === void 0) { prop = {}; }
            /*
            if (this.componentRefs[prop.id]){
                (<ContainerProperties>this.componentRefs[prop.id].instance).closeOverlay();
            }
            */
        };
        Overlay.prototype.applyPropertieDefaults = function (defaultProperties, properties) {
            if (!properties) {
                properties = {};
            }
            if (!properties.index) {
                properties.index = 0;
            }
            this._defaultProperties = Object.assign({}, defaultProperties);
            return Object.assign(this._defaultProperties, properties);
        };
        Overlay.prototype.objectLength = function (obj) {
            var length = 0, key;
            for (key in obj) {
                if (obj.hasOwnProperty(key))
                    length++;
            }
            return length;
        };
        ;
        Overlay.prototype.handleGlobalEvents = function (event) {
            if (event.type === '[Overlay] Hide') {
                this.handleCloseEvent();
            }
        };
        Overlay.prototype.handleCloseEvent = function () {
            //const id = 'popover'; // Note: pass id in event
            this.appRef.detachView(this.componentRefs[0].hostView);
            this.componentRefs[0].destroy();
            delete this.componentRefs[0];
        };
        Overlay.ctorParameters = function () { return [
            { type: core.ComponentFactoryResolver },
            { type: core.ApplicationRef },
            { type: core.Injector },
            { type: EventService }
        ]; };
Overlay.ɵfac = function Overlay_Factory(t) { return new (t || Overlay)(ɵngcc0.ɵɵinject(ɵngcc0.ComponentFactoryResolver), ɵngcc0.ɵɵinject(ɵngcc0.ApplicationRef), ɵngcc0.ɵɵinject(ɵngcc0.Injector), ɵngcc0.ɵɵinject(EventService)); };
Overlay.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: Overlay, factory: function (t) { return Overlay.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Overlay, [{
        type: core.Injectable
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }, { type: ɵngcc0.ApplicationRef }, { type: ɵngcc0.Injector }, { type: EventService }]; }, null); })();
        return Overlay;
    }());

    var defaultProperties$1 = {
        transitionDuration: 400,
        minSwipeDistance: 50,
        transitionTimingFunction: 'ease',
        counterSeparator: " / ",
        arrows: true
    };

    var Gallery = /** @class */ (function () {
        function Gallery(overlay, overlayEventService) {
            this.overlay = overlay;
            this.overlayEventService = overlayEventService;
        }
        Gallery.prototype.load = function (properties) {
            properties = this.applyPropertieDefaults(defaultProperties$1, properties);
            this.overlay.load({
                id: 'gallery',
                mainComponent: CarouselComponent,
                width: document.body.clientWidth + 'px',
                overlayBackdrop: true,
                position: 'fixed',
                metadata: {
                    images: properties.images,
                    index: properties.index,
                    width: document.body.clientWidth,
                    height: window.innerHeight,
                    objectFit: 'none',
                    margin: 0,
                    transitionDuration: properties.transitionDuration,
                    overflowCellsLimit: 1,
                    transitionTimingFunction: properties.transitionTimingFunction,
                    counter: properties.counter,
                    counterSeparator: properties.counterSeparator,
                    arrows: properties.arrows,
                    minSwipeDistance: properties.minSwipeDistance
                }
            });
        };
        Gallery.prototype.close = function () {
            this.overlayEventService.emitChangeEvent({
                type: 'Hide'
            });
        };
        Gallery.prototype.applyPropertieDefaults = function (defaultProperties, properties) {
            if (!properties) {
                properties = {};
            }
            for (var propertie in properties) {
                if (properties[propertie] === undefined) {
                    delete properties[propertie];
                }
            }
            this._defaultProperties = Object.assign({}, defaultProperties);
            return Object.assign(this._defaultProperties, properties);
        };
        Gallery.ctorParameters = function () { return [
            { type: Overlay },
            { type: EventService }
        ]; };
Gallery.ɵfac = function Gallery_Factory(t) { return new (t || Gallery)(ɵngcc0.ɵɵinject(Overlay), ɵngcc0.ɵɵinject(EventService)); };
Gallery.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: Gallery, factory: function (t) { return Gallery.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Gallery, [{
        type: core.Injectable
    }], function () { return [{ type: Overlay }, { type: EventService }]; }, null); })();
        return Gallery;
    }());

    var OverlayModule = /** @class */ (function () {
        function OverlayModule() {
        }
OverlayModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: OverlayModule });
OverlayModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function OverlayModule_Factory(t) { return new (t || OverlayModule)(); }, providers: [
        Overlay,
        EventService
    ], imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(OverlayModule, { declarations: [OverlayContainerComponent, AdDirective], imports: [ɵngcc1.CommonModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OverlayModule, [{
        type: core.NgModule,
        args: [{
                declarations: [
                    OverlayContainerComponent,
                    AdDirective
                ],
                imports: [
                    common.CommonModule
                ],
                exports: [],
                providers: [
                    Overlay,
                    EventService
                ],
                bootstrap: [],
                entryComponents: [
                    OverlayContainerComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return OverlayModule;
    }());

    var IvyGalleryModule = /** @class */ (function () {
        function IvyGalleryModule() {
        }
IvyGalleryModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IvyGalleryModule });
IvyGalleryModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IvyGalleryModule_Factory(t) { return new (t || IvyGalleryModule)(); }, providers: [
        Gallery
    ], imports: [[
            common.CommonModule,
            OverlayModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IvyGalleryModule, { declarations: [CarouselComponent], imports: [ɵngcc1.CommonModule, OverlayModule], exports: [CarouselComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IvyGalleryModule, [{
        type: core.NgModule,
        args: [{
                declarations: [
                    CarouselComponent
                ],
                imports: [
                    common.CommonModule,
                    OverlayModule
                ],
                exports: [
                    CarouselComponent
                ],
                providers: [
                    Gallery
                ],
                bootstrap: [],
                entryComponents: [
                    CarouselComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return IvyGalleryModule;
    }());

    exports.Gallery = Gallery;
    exports.IvyGalleryModule = IvyGalleryModule;
    exports.ɵa = CarouselComponent;
    exports.ɵb = EventService;
    exports.ɵc = OverlayModule;
    exports.ɵd = OverlayContainerComponent;
    exports.ɵe = AdDirective;
    exports.ɵf = Overlay;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
