// angular-gallery/esm5/lib/gallery.module.js
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CarouselComponent } from './carousel.component';
import { Gallery } from './gallery.service';
import { OverlayModule } from './overlay/overlay.module';
import * as ɵngcc0 from '@angular/core';
var IvyGalleryModule = /** @class */ (function () {
    function IvyGalleryModule() {
    }
IvyGalleryModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IvyGalleryModule });
IvyGalleryModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IvyGalleryModule_Factory(t) { return new (t || IvyGalleryModule)(); }, providers: [
        Gallery
    ], imports: [[
            CommonModule,
            OverlayModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IvyGalleryModule, { declarations: function () { return [CarouselComponent]; }, imports: function () { return [CommonModule,
        OverlayModule]; }, exports: function () { return [CarouselComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IvyGalleryModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    CarouselComponent
                ],
                imports: [
                    CommonModule,
                    OverlayModule
                ],
                exports: [
                    CarouselComponent
                ],
                providers: [
                    Gallery
                ],
                bootstrap: [],
                entryComponents: [
                    CarouselComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return IvyGalleryModule;
}());
export { IvyGalleryModule };
